nifi-nar-bundles/nifi-extension-utils/nifi-database-utils/src/test/java/org/apache/nifi/util/db/TestJdbcCommon.java

```diff
@@ -84,23 +83,20 @@
     private static final Logger LOGGER = LoggerFactory.getLogger(TestJdbcCommon.class);
     static final String createTable = "create table restaurants(id integer, name varchar(20), city varchar(50))";
 
-    @ClassRule
-    public static TemporaryFolder folder = new TemporaryFolder();
-
     /**
      * Setting up Connection is expensive operation.
      * So let's do this only once and reuse Connection in each test.
      */
     static protected Connection con;
 
-    @BeforeClass
-    public static void setup() throws ClassNotFoundException, SQLException {
+    @BeforeAll
+    public static void setup() throws ClassNotFoundException, SQLException, IOException {
         System.setProperty("derby.stream.error.file", "target/derby.log");
         Class.forName("org.apache.derby.jdbc.EmbeddedDriver");
-        // remove previous test database, if any
-        folder.delete();
 
-        String location = folder.getRoot().getAbsolutePath();
+        String location = Files.createTempDirectory(String.valueOf(System.currentTimeMillis()))
+                .resolve("db")
+                .toFile().getAbsolutePath();
```

Comment:
As mentioned in related tests, see `DBCPServiceTest` for an approach to handle database directory creation and clearing.

Reply:
This test class still needs a method to clear the temporary database directory after test method completion.